### ISSUE TITLE
Link underlining is inconsistent, not robust for accessibility

### ISSUE THREAD

**Issue by fuzzbomb** [thememachine/eleven#158]
# Problem

Links inside of surrounding text (sentences) have underlines, which is AWESOME and helps satisfy [WCAG SC 1.4.1 Use of Color](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html).

However not all links inside surrounding text have underlines, and there are several styles:
- No underline - default style from `a { text-decoration: none; }`
- Solid underline - e.g. from `.messages a { text-decoration: underline;}`
- Dashed underline - from `p a, .content a { `
- Solid and Dashed underline (two lines) - happens when a link matches two selectors.

This creates confusion - doe the different link styles convey a different meaning?

# Where

- Menu UI help page - `admin/help/menu_ui`
  This has a mixture of dashed underlines and no underlines.
  ![eleventheme-mixed-link-menu-ui-help-page](https://user-images.githubusercontent.com/1243005/28934647-9f1534d2-7879-11e7-8142-6c79ec2c22bd.png)

- Getting started box - `admin/help`
  This has a mix of solid underline, and solid-dashed-double underline
  Solid underline comes from `.message a {text-decoration: underline }`.
  The solid-and-dashed double underline is due to matching `.messages a` and `p a` selectors.
  ![eleventheme-mixed-links-help-landing-page](https://user-images.githubusercontent.com/1243005/28934752-fb00bfaa-7879-11e7-80f4-875c906a756c.png)

 # Recommendation

1. Use a single underline for links inside surrounding text.  Make this the DEFAULT style, i.e. `a { text-decoration: underline; }`.   This is super-highly recommended for accessibility, and is the most reliable way to avoid [WCAG F73: Failure of Success Criterion 1.4.1 due to creating links that are not visually evident without color vision](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F73).  It's also a long-standing convention, and it's how the web looks before designers break it ;-)  See also:
    - [On Link Underlines](http://adrianroselli.com/2016/06/on-link-underlines.html)
    - [WebAxe: Keep the Underline](http://www.webaxe.org/keep-the-underline-text-links/)

2. Get rid of lots of other selectors, such as `.messages a` and `pa, .content a`.  You won't need them once you've done the previous steps. 

3. For links that are not inside surrounding text, AND have some other visual affordance (e.g. they are items in a toolbar), it's safe to override with a more specific selector like `.toolbar a`.

4. [aside] - if you want to use an alternative fancy link style, like underlining in a different colour, the new CSS4 [`text-decoration-*`](https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration-color) are the preferred way, but there's no MSIE support.  Note that `border` is much more robust than `box-shadow`.  This is because some assistive tech which provides alternative colour spaces will selectively override/ignore some CSS.  For example, Windows High Contrast mode respects border style + width, and overrides border-color (so the border is still seen).  However Windows High Contrast causes  `box-shadow` to be stripped out entirely, so it won't be seen.

**Comment by cehfisher** [comment]
Looks so much better! Thanks for making these changes @mortendk. Here is one link you missed. Ex. page: http://drupalvm-eleven.dev/admin/appearance

![underline](https://user-images.githubusercontent.com/15240149/29221472-8023a4ee-7e84-11e7-9716-0f0a46e30b11.png)

**Comment by mortendk** [comment]
this link @cehfisher 
![image](https://user-images.githubusercontent.com/65756/29710956-dd315e7e-8992-11e7-929e-5abfc59baff1.png)
